nucypher/blockchain/eth/sol/source/contracts/PolicyManager.sol

```diff
@@ -374,42 +449,83 @@ contract PolicyManager is Upgradeable {
     }
 
     /**
-    * @notice Revoke policy by client
+    * @notice Revoke policy by the creator
     * @param _policyId Policy id
     */
-    function revokePolicy(bytes16 _policyId) public {
-        refundInternal(_policyId, RESERVED_NODE, true);
+    function revokePolicy(bytes16 _policyId) public returns (uint256 refundValue) {
+        require(getPolicyOwner(_policyId) == msg.sender);
+        return refundInternal(_policyId, RESERVED_NODE, true);
     }
 
     /**
-    * @notice Revoke arrangement by client
+    * @notice Revoke arrangement by the creator
     * @param _policyId Policy id
     * @param _node Node that will be excluded
     */
     function revokeArrangement(bytes16 _policyId, address _node)
         public returns (uint256 refundValue)
     {
         require(_node != RESERVED_NODE);
+        require(getPolicyOwner(_policyId) == msg.sender);
         return refundInternal(_policyId, _node, true);
     }
 
     /**
-    * @notice Refund part of fee by client
+    * @notice Get unsigned hash for revocation
+    * @param _policyId Policy id
+    * @param _node Node that will be excluded
+    * @return Revocation hash
+    */
+    function getRevocationHash(bytes16 _policyId, address _node) public view returns (bytes32) {
+        return SignatureVerifier.hashEIP191(abi.encodePacked(_policyId, _node), byte(0x45));
```

Comment:
I think this line deserves a comment explaining we're using EIP191 version `0x45 ('E')` for `personal_sign`

Reply:
✔️